### DISCUSSION TITLE
How Monaco implements search highlighting?

### DISCUSSION THREAD

**Discussion by yjhtry** [microsoft/monaco-editor#4246] [Q&A] [answered]
I have a same requirement in my application and I cannot use Monaco directly. So I highlighted the search terms by modifying the DOM. It performed well when there were few nodes, but as the number of nodes increased, the program became more and more stuck.

I saw that Monaco did not change the dom, and the search did not become laggy with the highlighting. Can any developer tell Monaco how to implement the change function? I would be very grateful.

![image](https://github.com/microsoft/monaco-editor/assets/39440537/03c5f78e-d777-4c48-8910-f09fe4b7f4f2)

**Comment by loggerhead** [comment]
@yjhtry I'm curious too, do you know the implementation is now?

**Reply by yjhtry** [reply, marked as answer]
Yes, look here [css highlight](https://developer.mozilla.org/zh-CN/docs/Web/CSS/::highlight)
